src/proposal.mligo

```diff
@@ -97,19 +120,15 @@ let stake_tk(token_amount, addr, period, store : nat * address * period * storag
   in { store with freeze_history = new_freeze_history; quorum_threshold_at_cycle = {store.quorum_threshold_at_cycle with staked = new_cycle_staked } }
 
 [@inline]
-let unfreeze_on_ledger (tokens, addr, ledger, total_supply, frozen_token_id, governance_token : nat * address * ledger * total_supply * token_id * governance_token): (operation * ledger * total_supply) =
+let unfreeze_on_ledger (tokens, addr, governance_token : nat * address * governance_token): operation =
   // Call transfer on token_contract to transfer `token` number of
   // tokens from `addr` to the address of this contract.
   let param = { from_ = Tezos.self_address; txs = [{ amount = tokens; to_ = addr; token_id = governance_token.token_id }]} in
   let operation = make_transfer_on_token ([param], governance_token.address) in
+  operation
 
-  // Once this contract is credited on the token contract, we can burn frozen tokens
-  // to debit the `addr` address here.
-  let (ledger, total_supply) = debit_from (tokens, addr, frozen_token_id, ledger, total_supply) in
-  (operation, ledger, total_supply)
-
-let add_proposal (propose_params, period, store : propose_params * period * storage): storage =
-  let proposal_key = ensure_proposal_is_unique (propose_params, store) in
+let add_proposal (from_, propose_params, period, store : address * propose_params * period * storage): storage =
```

Comment:
Why is `from_` needed? Can't we use `propose_param.from_`?

Reply:
Not really needed, I've removed this now.